readme.md

```diff
@@ -1,3 +1,103 @@
-# PyTest energy reporter
+# pytest energy reporter
 
-TODO
\ No newline at end of file
+The pytest energy reporter can be used to report on energy usage of code, by measuring tests.
+
+## Installation
+
+To install pytest energy reporter, simply run:
+
+```bash
+pip install pytest_energy_reporter
+```
+
+## Usage
+
+The package is automatically enabled in pytest. To start measuring energy consumption, mark tests as energy test using:
+
+```python
+import pytest
+from pytest_energy_reporter.measurement import measure_energy
+
+@pytest.mark.energy
+def test_fn():
+
+# Or define the number of iterations to measure
+@pytest.mark.energy(n=3)
+def test_fn():
+
+# Or use the measurement directly
+def test_fn():
+  result = measure_energy(fn)
+  assert result.energy_j < 200
+
+
+```
+
+The plugin exposes a few flags in pytest
+
+| Flag | Argument | Meaning |
+| --- | --- | --- |
+| `--save-energy-report` | `<none>` | Will save the energy report in `reports/energy/` 
+| `--energy-iterations` | `int` | Specify the number of iterations for the energy measurement. |
+
+## Contributing
+
+### Prerequistes
+
+Make sure you have [poetry](https://python-poetry.org/) installed.
+
+### Installation
+
+To install the pytest energy reporter repository, run:
+
+1. Clone the repository
+```bash
+git clone git@github.com:delanoflipse/pytest-energy-reporter.git
+```
+
+2. Clone subrepositories
+
+```bash
+git submodule init && git submodule update
+```
```

Comment:
The file `.gitmodules` specifies the SSH clone link. I would suggest to use the HTTPS clone link so people do not have to setup their public keys but can clone the repo without any setting up.